executor/inspection_result.go

```diff
@@ -459,6 +470,63 @@ func (criticalErrorInspection) inspect(ctx context.Context, sctx sessionctx.Cont
 	return results
 }
 
+func (criticalErrorInspection) inspectForServerDown(ctx context.Context, sctx sessionctx.Context, filter inspectionFilter) []inspectionResult {
+	item := "server-down"
+	if !filter.enable(item) {
+		return nil
+	}
+	condition := filter.timeRange.Condition()
+	sql := fmt.Sprintf(`select t1.job,t1.instance, t2.min_time from
+		(select instance,job from metrics_schema.up %[1]s group by instance,job having max(value)-min(value)>0) as t1 join
+		(select instance,min(time) as min_time from metrics_schema.up %[1]s and value=0 group by instance,job) as t2 on t1.instance=t2.instance order by job`, condition)
+	rows, _, err := sctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQLWithContext(ctx, sql)
+	if err != nil {
+		sctx.GetSessionVars().StmtCtx.AppendWarning(fmt.Errorf("execute '%s' failed: %v", sql, err))
+	}
+	var results []inspectionResult
+	for _, row := range rows {
+		if row.Len() < 3 {
+			continue
+		}
+		detail := fmt.Sprintf("%s %s disconnect with prometheus around time '%s'", row.GetString(0), row.GetString(1), row.GetTime(2))
+		result := inspectionResult{
+			tp:       row.GetString(0),
+			instance: row.GetString(1),
+			item:     item,
+			actual:   "",
+			expected: "",
+			severity: "critical",
+			detail:   detail,
+			degree:   10000 + float64(len(results)),
+		}
+		results = append(results, result)
+	}
+	// Check from log.
+	sql = fmt.Sprintf("select type,instance,time from information_schema.cluster_log %s and level = 'info' and message like '%%Welcome to'", condition)
+	rows, _, err = sctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQLWithContext(ctx, sql)
+	if err != nil {
+		sctx.GetSessionVars().StmtCtx.AppendWarning(fmt.Errorf("execute '%s' failed: %v", sql, err))
+	}
+	for _, row := range rows {
+		if row.Len() < 3 {
+			continue
+		}
+		detail := fmt.Sprintf("%s %s restarted at time '%s'", row.GetString(0), row.GetString(1), row.GetString(2))
+		result := inspectionResult{
+			tp:       row.GetString(0),
+			instance: row.GetString(1),
+			item:     item,
+			actual:   "",
+			expected: "",
+			severity: "critical",
+			detail:   detail,
+			degree:   10000 + float64(len(results)),
```

Comment:
Any reason for assign `degree` here?

Reply:
As the comment of degree said, 
```
		// degree only used for sort.
``` 
, assign a value here only use to sort the result, it is an important message, so we need to put this message in the front of the result.